src/system.rs

```diff
@@ -75,6 +90,13 @@ pub trait SystemData<'a> {
     ///
     /// Only fetch the resources you
     /// returned from `reads` / `writes`!
+    /// This function may panic if this contract is violated.
```

Comment:
This seems redundant to me, but you can leave if you feel it adds to the documentation.